src/utils/querybuilder.util.js

```diff
@@ -514,16 +515,49 @@ const quantityQueryBuilder = function ({ target, field }) {
                 qB[`${field}.value`] = { $gte: num };
                 break;
             case 'ne':
-                qB[`${field}.value`] = { $ne: num };
+                offset = calcRange(num);
+                num = Number(num);
+                qB[`${field}.value`] = { $ne: { $gt: num - offset, $lt: num + offset }};
                 break;
         }
     } else {
         // no prefixes
-        qB[`${field}.value`] = { $gt: 0.99 * num, $lt: 1.01 * num };
+        offset = calcRange(num);
+        num = Number(num);
+        qB[`${field}.value`] = { $gt: num - offset, $lt: num + offset };
     }
 
     return qB;
 };
+// return value of range offset
+function calcRange(numStr) {
```

Comment:
Handling for case of '1e2' is missing, the range should be {'$gte':'95','$lt':'105'} but currently it is {'$gt':'99.5','$lt':'100.5'}

Reply:
Fixed scientific notation range generation